semantiva/data_processors/parametric_sweep_factory.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2025 Semantiva authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, Dict, Tuple, Type, Sequence
+
+import numpy as np
+
+from semantiva.data_io.data_io import DataSource
+from semantiva.data_processors.data_processors import DataOperation
+from semantiva.data_types import NoDataType, DataCollectionType
+
+
+class ParametricSweepFactory:
+    """Factory for creating parametric sweep data operations."""
```

Comment:
This docstring is too vague. We need an objetive statement of what this Factory produces. The class created by this factory takes a DataSource and produces a DataCollection of the respective output data type.